HGustavs/LenaSYS
# Codeexample sections now shows up under the correct moment. g3-2024-v6-#15319

Extra:
* Added .vscode to .gitignore to make it easier to debug using visual studio code.
* Fixed errors occurring in sectioned.php due to them obscuring important footer texts.
* Fixed sections not inserting with correct positions or duplicate positions which would break functionality.

### Testing
1. Create a empty test course:
2. Connect https://github.com/LenaSYS/Webbprogrammering-Examples as github URL
3. Create a course version and add 7 Moments with some few random elements sprinkled in. (ajax_api_index, booking and README have a chance of showing up after refreshing moments without choosing anything. The bug can be found before my issue.)
![image](https://github.com/HGustavs/LenaSYS/assets/102598258/66074118-a0b6-4317-a539-f3b851034cb8)

4. Ensure that the positions `pos` are correct in `listentries` in the database
![image](https://github.com/HGustavs/LenaSYS/assets/102598258/f0382ce6-f89f-4bb7-8fc2-cf881a6626d9)
5. (One at a time) Choose GitHub moment and then press the refresh moment example icon on the moment. After a site refresh the codeexamples should pop up.
![image](https://github.com/HGustavs/LenaSYS/assets/102598258/3ba827af-6b0d-46b6-b823-899f733f7ffe)
![image](https://github.com/HGustavs/LenaSYS/assets/102598258/ff90df20-13d4-47cf-ac61-28958136bc74)
6. Verify  `pos` in the database
